### DISCUSSION TITLE
Methods to Handle Request Type

### DISCUSSION THREAD

**Discussion by devajmeireles** [lorisleiva/laravel-actions#146] [Q&A] [answered]
Hey, there!

I started using InertiaJS. Is there any way an Action can support a "GET" method and another "POST/PUT/DELETE" method?

I tried using the scheme:

1. asController
2. htmlResponse
3. jsonResponse

... But both requests (GET: page access and POST/PUT: data sent by InertiaJS) are sent to `asController`. It would be interesting to be able to separate this into different methods, such as:

`asControllerGet`
`asControllerPost`
`asControllerPut`

... And so on!

C/C @lorisleiva

**Comment by lorisleiva** [comment, marked as answer]
Hi there 👋 

Does this help? https://laravelactions.com/2.x/register-as-controller.html#routes-with-explicit-methods

**Reply by devajmeireles** [reply]
@lorisleiva I hadn't seen that part! Thank you.

Taking the opportunity. How can I sponsor you with an amount less than $ 4?

**Reply by lorisleiva** [reply]
No worries! Not on GitHub but I've got a ["Buy me a coffee" account](https://www.buymeacoffee.com/loris) for custom one time donations if you want. ☺️
